cel/src/context.rs

```diff
@@ -33,10 +34,12 @@ pub enum Context<'a> {
     Root {
         functions: FunctionRegistry,
         variables: HashMap<String, Value>,
+        resolver: Option<Arc<dyn VariableResolver>>,
```

Comment:
Thought process behind requiring this to be `Arc`d? Why not just `Option<&'a dyn VariableResolver>`?

Reply:
I am not sure the `'a` lifetime will work well. At least for my usage, I have `Context<'static>` while my variable resolver is not `'static`. We could add a different lifetime but then its leaking everywhere

Reply:
Pushed up this change